src/build.py

```diff
@@ -611,9 +611,16 @@ def __init__(self, include=None, exclude=None):
 
   def Apply(self, targets):
     """ Return the filtered list of targets. """
+    all_names = [t.name for t in targets]
     if self.include is not None:
+      missing_includes = [i for i in self.include if i not in all_names]
+      if missing_includes:
+        raise Exception('Invalid step name(s): ' + str(missing_includes))
       return [t for t in targets if t.name in self.include]
     if self.exclude:
+      missing_excludes = [i for i in self.include if i not in all_names]
+      if missing_excludes:
+        raise Exception('Invalid step name(s): ' + str(missing_excludes))
```

Comment:
I like that pretty well. I might call it `specified_names` instead of `specified_list`.